environments/gcp/flyte-core/iam.tf

```diff
@@ -0,0 +1,174 @@
+  
+}
+
+resource "google_project_iam_binding" "flytescheduler-binding" {
+  project = local.project_id
+  role = google_project_iam_custom_role.custom_IAM_roles["flytescheduler"].id
+  members =["serviceAccount:${google_service_account.flytescheduler_gsa.email}"]
+  
+}
+
+resource "google_project_iam_binding" "datacatalog-binding" {
+  project = local.project_id
+  role = google_project_iam_custom_role.custom_IAM_roles["datacatalog"].id
+  members = ["serviceAccount:${google_service_account.datacatalog_gsa.email}"]
+  
+}
+
+resource "google_project_iam_binding" "flyteworkers-binding" {
+  project = local.project_id
+  role = google_project_iam_custom_role.custom_IAM_roles["flyteworkers"].id
+  members = ["serviceAccount:${google_service_account.flyteworkers_gsa.email}"]
+  
+}
+
+# Step 4 Bind GSAs with KSAs as Workload Identity Users, enabling impersonation
+resource google_service_account_iam_binding "flyteadmin_workload_identity_binding" {
+   role               = "roles/iam.workloadIdentityUser"
+   service_account_id = google_service_account.flyteadmin_gsa.name
+   members = ["serviceAccount:${module.gke.identity_namespace}[flyte/flyteadmin]"]
+
+}
+
+resource google_service_account_iam_binding "flytepropeller_workload_identity_binding" {
+   role               = "roles/iam.workloadIdentityUser"
+   service_account_id = google_service_account.flytepropeller_gsa.name
+   members = ["serviceAccount:${module.gke.identity_namespace}[flyte/flytepropeller]"]
+
+}
+
+resource google_service_account_iam_binding "flytescheduler_workload_identity_binding" {
+   role               = "roles/iam.workloadIdentityUser"
+   service_account_id = google_service_account.flytescheduler_gsa.name
+   members = ["serviceAccount:${module.gke.identity_namespace}[flyte/flytescheduler]"]
+
+}
+
+resource google_service_account_iam_binding "datacatalog_workload_identity_binding" {
+   role               = "roles/iam.workloadIdentityUser"
+   service_account_id = google_service_account.datacatalog_gsa.name
+   members = ["serviceAccount:${module.gke.identity_namespace}[flyte/datacatalog]"]
+
+}
+
+
+
+locals {
+  flyte_worker_wi_members = toset([
+    for tpl in setproduct(
+      local.flyte_projects,
+      local.flyte_domains,
+      local.flyte_ksas
+    ) : format("%s-%s/%s", tpl...)
+  ])
+}
+
+# IAM Policy for the Flyte workers role. The SA IAM Policy resource is authoritative for, in this case, the Flyteworkers SA. 
+#It is implemented in this way both as a convenience to be able to configure SA impersonation for the `default` SA of 
+#each and every project-domain combination in a programmatic way, but also because the Flyteworkers SA should only act as a
+#Workload Identity User. 
+
+data "google_iam_policy" "flyte_worker_workload_identity" {
+  binding {
+    members = formatlist("serviceAccount:${module.gke.identity_namespace}[%s]", local.flyte_worker_wi_members)
+    role    = "roles/iam.workloadIdentityUser"
+  }
```

Comment:
I haven't used this syntax to configure workload identities. I would have expected the a `service_account_id` arg here as in the `google_service_account_iam_binding` resources above.

Not saying this doesn't work, just sanity check please.